src/NumberTheory/vinberg.jl

```diff
@@ -353,3 +353,163 @@ function vinberg_algorithm(S::ZZLat, upper_bound; v0=QQ[0;]::QQMatrix, root_leng
+
+Return a tuple of a boolean that represents if isometry f of the lattice is positive and QQMatrix,
+that represents an obstructing root
+"""
+function _process_finite_sets_of_h(h, f::QQMatrix, v, w, bi_form, L::ZZLat):: Tuple{Bool, QQMatrix}
+  x = bi_form(h, h)
+  y = bi_form(h, h*f)
+  z = y^2-x^2
+  if (z<0) return (true, zero(h)) end #then discrininant for a will be <0 and there is no roots (a,b)=> no obstructing roots => positive
+  #trunc(ZZRingElem,-sqrt(2*z/x)) #check issues on github as round works in the same context
+  b_min = round(ZZRingElem, -sqrt(2*z/x), RoundUp)
+  for b = b_min:-1
+    a_max = round(ZZRingElem, (ZZ(b)*y+sqrt(z*(ZZ(b)^2+2*x)))/x, RoundDown)
+    for a = 1:a_max
+      h_new = -ZZ(b)*h + ZZ(a)*(h*f)
+      Rh = _get_R(L, h_new)
+      for r in Rh
+        result = _check_R(r, v, w, bi_form)
+        if !result[1] 
+          return result
+        end
+      end
+    end
+  end
+  return (true, zero(h))
+end
+
+function _check_R(r, v, w, bi_form) :: Tuple{Bool, QQMatrix}
+  if bi_form(r, v)*bi_form(r, w) < 0 return (false, r)
+  else return (true, zero(r)) end
+end
+
+@doc raw"""
+  isometry_is_positive(Lf::ZZLatWithIsom, h::Union{QQMatrix, Nothing} = nothing) -> Tuple{Bool, QQMatrix}
+
+Given lattice with isometry 'Lf' and given 'h' vector in this lattice
+according to the ambient space basis, such that $h^2>0$. 
+If no such vector is given, 'h' will be calculated by function inside based on a random vector in 'Lf'.
+
+Return a tuple of a boolean that represents if isometry f of the lattice is positive and QQMatrix,
+that represents an obstructing root (see Algorithm 5.8 of "MINIMUM POSITIVE ENTROPY OF COMPLEX
+ENRIQUES SURFACE AUTOMORPHISMS" by KEIJI OGUISO and XUN YU in Duke Mathematical Journal, 2020,
+volume 169, number 18, pages 3565 - 3606)
+
+For positive isometries it returns a vector of zeros with the same dimension as 'h'.
+"""
+
+function isometry_is_positive(Lf::ZZLatWithIsom, h::Union{QQMatrix, Nothing} = nothing):: Tuple{Bool, QQMatrix}
+  Qb = algebraic_closure(QQ);
+  f = ambient_isometry(Lf)
+  L = lattice(Lf)
+  tau = _get_tau(f, Qb)
+  bi_form = _get_bilinealform(L, Qb)
+
+  # step 1
+  C0 = _get_C0(Lf, tau)
+  # step 2 - Check if C0 has obstructing roots => positive
+  if !isone(C0)
+    Cfancy = _get_Cfancy(Lf, C0)
+    first_element = iterate(Cfancy)
+    if first_element !== nothing
+      return (false, first_element[1][1])
+    end
+  end
+  # step 3 - Prepare eigenvectors from tau and tau inverse 
+  v = _get_eigenvector(f, tau)
+  w = _get_eigenvector(f, tau^(-1))
+
+  if bi_form(v,w)<0
+    v = -v
+  end
+  # step 4 - Get the first h value if there is no in arguments
+  if (h===nothing)
+    h = _get_h(L,v,w, Qb, bi_form)
+  end
+  # step 5 - Get the R set based on current h value
+  Rh = _get_R(L, h)
+  # step 6 - Check all of the entries of R if there exists obstructing root => positive
+  for r in Rh
+    result = _check_R(r, v, w, bi_form) 
+    if !result[1] return result end
```

Comment:
As above, with my suggested change this could be come
```suggestion
    _check_R(r, v, w, bi_form) || return false, r
```